scenario14/jmeter/01-Scenario14-EnforcePasswordReset.jmx

```diff
@@ -824,394 +839,554 @@
+import com.gargoylesoftware.htmlunit.*;
+import com.gargoylesoftware.htmlunit.BrowserVersion;
+
+try {
+		public HtmlUnitDriver driver = new HtmlUnitDriver(BrowserVersion.CHROME, true);
+
+		String protocole = &quot;http://&quot;;
+		String tomcathost = &quot;${tomcatHost}&quot;;
+		String seperator = &quot;:&quot;;
+		String tomcatport = &quot;${tomcatPort}&quot;;
+		String context = &quot;/travelocity.com/index.jsp&quot;;
+        
+
+	    // String url=vars.get(&quot;Location2&quot;);
+
+
+		String AppURL = protocole + tomcathost + seperator + tomcatport + context;
+          		
+		driver.get(AppURL);
+		driver.manage().timeouts().implicitlyWait(15,TimeUnit.SECONDS);
+          
+
+		String pageSource = driver.getPageSource();
+
+         //Click on the link with the text as hear
+          WebElement hear = driver.findElement(By.xpath(&quot;/html/body/div/div[2]/div[2]/h2[1]/a&quot;));
+		hear.click();        
+	     
+      	WebElement username =  driver.findElement(By.id(&quot;username&quot;));
+		username.clear();
+          username.sendKeys(new String[] { vars.get(&quot;userNamePrefix&quot;) });
+
+		WebElement password =  driver.findElement(By.id(&quot;password&quot;));
+		password.clear();
+          password.sendKeys(new String[] { vars.get(&quot;userPassword&quot;) });
+		
+
+		WebElement button = driver.findElement(By.xpath(&quot;/html/body/div/div/div/div/div[2]/div[2]/form/div[6]/div/button&quot;));
+		
+		button.click();
+		
+
+		driver.manage().timeouts().implicitlyWait(15,TimeUnit.SECONDS);
+		
+
+		  
+          WebElement username =  driver.findElement(By.id(&quot;currentPassword&quot;));
+		username.clear();
+          username.sendKeys(new String[] { vars.get(&quot;userPassword&quot;) });
+
+		WebElement password =  driver.findElement(By.id(&quot;newPassword&quot;));
+		password.clear();
+          password.sendKeys(new String[] { vars.get(&quot;newPassword&quot;) });
+
+		WebElement password2 =  driver.findElement(By.id(&quot;repeatPassword&quot;));
+		password2.clear();
+          password2.sendKeys(new String[] { vars.get(&quot;newPassword&quot;) });
+
+		WebElement button = driver.findElement(By.id(&quot;changepass&quot;));
+		button.click();
+		
+		return driver.getPageSource();      	
+
+
+
+		 
+		  		
+		
+    
+} catch (Exception ex) {
+    ex.printStackTrace();
+    log.error(ex.getMessage());
+     SampleResult.setSuccessful(false);
+     SampleResult.setResponseCode(&quot;500&quot;);
+     SampleResult.setResponseMessage(ex.getMessage());
+     
+} </stringProp>
+          <stringProp name="scriptLanguage">java</stringProp>
+        </JSR223Sampler>
         <hashTree>
           <ResponseAssertion guiclass="AssertionGui" testclass="ResponseAssertion" testname="Response Assertion" enabled="true">
             <collectionProp name="Asserion.test_strings">
-              <stringProp name="1772522522">You are logged in as sol14user</stringProp>
+              <stringProp name="2524">OK</stringProp>
```

Comment:
Better to assert the expected results rather than 'OK' response.

Reply:
thanks for the reply, I have updated with the changes